src/components/RestaurantFilterComponent/RestaurantFilterComponent.jsx

```diff
@@ -1,260 +1,131 @@
 import { useSelector } from "react-redux";
 import "./RestaurantFilterComponent.css";
 import { useEffect } from "react";
-import { BsFillFilterSquareFill } from "react-icons/bs";
 import { useState } from "react";
-import { ImInfo } from "react-icons/im";
 import { useSearchParams } from "react-router-dom";
 import languageSelector from "../../assets/languages/languageSelector";
+import burger from "../../assets/hamburger.svg";
+import CheckboxFilter from "./CheckboxFilter";
+import cuisinesDB from "../../assets/cuisines.json";
 
-const RestaurantFilterComponent = ({
-  categories,
-  setCategories,
-  rating,
-  setRating,
-  data,
-}) => {
+const RestaurantFilterComponent = () => {
+  const cuisines = Object.keys(cuisinesDB);
   const [searchParams, setSearchParams] = useSearchParams({});
 
-  const [mobileFilter, setMobileFilter] = useState("");
   const [radioSelected, setRadioSelected] = useState("");
-  const [checkBoxSelected, setCheckBoxSelected] = useState([]);
+  const [mobileView, setMobileView] = useState(false);
 
   const language = useSelector((state) => state.languageReducer);
 
+  //the existence of a rating type filter in the url is verified, if so, the corresponding value is applied to the state of the radio buttons so that it is shown selected
   useEffect(() => {
-    const handleResize = () => {
-      if (window.innerWidth >= 765) {
-        setMobileFilter("");
-      }
-    };
-    window.addEventListener("resize", handleResize);
-
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
+    if (Object.fromEntries(searchParams.entries()).rating) {
+      setRadioSelected(Object.fromEntries(searchParams.entries()).rating);
+    }
   }, []);
 
-  // fires when a categorie is selected, adds the value to he state checkboxSelected and updates the searchParam with whats comming from the cuisine arrray and keeps the other values if they exist.
-  const handleCategoriesChange = (event) => {
-    if (event.target.checked) {
-      setCheckBoxSelected([...checkBoxSelected, event.target.name]);
+  // This function is executed from the checkboxes component, it receives the name of the checkbox and its value, if the value is "true" it adds the value to the searchParams, otherwise it removes it
+  const handleCategoriesChange = (element, value) => {
```

Comment:
```
/**
  This function is executed from the checkboxes component, 
  it receives the name of the checkbox and its value,
   if the value is true it adds the value to the searchParams, 
   otherwise it removes it.
*/
```

Reply:
Thinking better, are necessary these comments?